kedro-datasets/kedro_datasets/snowflake/snowpark_dataset.py

```diff
@@ -0,0 +1,225 @@
+"""``AbstractDataSet`` implementation to access Snowflake using Snowpark dataframes
+"""
+import logging
+from copy import deepcopy
+from typing import Any, Dict, Union
+
+import pandas as pd
+import snowflake.snowpark as sp  # pylint: disable=import-error
+from kedro.io.core import AbstractDataSet, DataSetError
+
+logger = logging.getLogger(__name__)
+
+
+class SnowParkDataSet(AbstractDataSet[pd.DataFrame, pd.DataFrame]):
```

Comment:
```suggestion
class SnowparkDataSet(AbstractDataSet[pd.DataFrame, pd.DataFrame]):
```
Snowpark, not SnowPark